libs/ui-watt/src/lib/components/button/watt-button.component.spec.ts

```diff
@@ -14,50 +14,90 @@
+  WattButtonVariant,
+} from './watt-button.component';
 import { WattButtonModule } from './watt-button.module';
 
-describe(WattButtonComponent.name, () => {
-  test.each([
-    ['text', WattTextButtonComponent],
-    ['primary', WattPrimaryButtonComponent],
-    ['secondary', WattSecondaryButtonComponent],
-  ] as readonly [WattButtonVariant, Type<unknown>][])(
-    'renders a %s button',
-    async (buttonVariant, buttonComponentType) => {
-      const view = await render(WattButtonComponent, {
-        componentProperties: {
-          variant: buttonVariant,
-        },
-        imports: [WattButtonModule],
-      });
-
-      const component = view.fixture.componentInstance;
-      expect(component.buttonComponentVariant).toBe(buttonComponentType);
-    }
-  );
+interface WattButtonAltOptions {
+  icon?: WattIcon;
+  loading?: boolean;
+  disabled?: boolean;
+  text?: string;
+  variant?: WattButtonVariant;
+  size?: WattButtonSize;
+}
+
+describe('WattButtonAltComponent', () => {
+  const renderComponent = async ({ ...options }: WattButtonAltOptions) => {
+    await render(
+      `<watt-button
+        variant=${options.variant}
+        size=${options.size}
+        icon=${options.icon}
+        [loading]=${options.loading}
+        [disabled]=${options.disabled}>
+          ${options.text ?? 'Text'}
+      </watt-button>`,
+      { imports: [WattButtonModule] }
+    );
+  };
+
+  it('shows text', async () => {
+    await renderComponent({ text: 'Text' });
+
+    expect(screen.getByRole('button')).toHaveTextContent('Text');
+  });
+
+  it('shows icon', async () => {
+    await renderComponent({ icon: 'plus' });
+
+    expect(screen.getByRole('img')).toHaveClass('mat-icon');
+  });
+
+  it('shows loading spinner', async () => {
+    await renderComponent({ loading: true });
+
+    expect(screen.getByRole('progressbar')).toHaveClass('mat-spinner');
+  });
 
-  test.each([undefined, null, ''])(
-    '`defaults to a text button (variant="$bottomValue")',
-    async (bottomValue) => {
-      const view = await render(WattButtonComponent, {
-        componentProperties: {
-          // intentionally pass bottom values
-          // eslint-disable-next-line @typescript-eslint/no-explicit-any
-          variant: bottomValue as any,
-        },
-        imports: [WattButtonModule],
-      });
-
-      const component = view.fixture.componentInstance;
-      expect(component.variant).toBe('text');
+  test.each(WattButtonTypes)(
+    'gets variant "%s" applied as class',
+    async (variant) => {
+      await renderComponent({ variant, text: 'Text' });
+
+      if (variant === 'icon') {
+        expect(screen.getByRole('button')).not.toHaveTextContent('Text');
+      }
+      expect(screen.getByRole('button')).toHaveClass('watt-button--' + variant);
     }
   );
+
+  it('gets size applied as class', async () => {
+    await renderComponent({ size: 'large' });
+
+    expect(screen.getByRole('button')).toHaveClass('watt-button--large');
+  });
```

Comment:
suggestion: add a similar test for `normal` size

Reply:
Fixed :)